src/pudl/transform/ferc1.py

```diff
@@ -1260,8 +1285,9 @@ def select_dbf_rows_by_category(
             )
         return processed_dbf
 
-    @cache_df(key="process_xbrl_metadata")
-    def process_xbrl_metadata(self: Self, xbrl_metadata_json) -> pd.DataFrame:
+    @property
+    @cache_df(key="pre_process_xbrl_metadata")
+    def pre_process_xbrl_metadata(self: Self) -> pd.DataFrame:
```

Comment:
i had to break up `process_xbrl_metadata` into THREE steps:
- `preprocess_xbrl_metadata`
- `process_calculation_components`
- `process_xbrl_metadata` bc this one needs the previous two.

Reply:
questions:
- names: `xbrl_metadata`, `xbrl_calculations` and `xbrl_metadata_raw`?? (its not really raw?)
 - should these all be properties? or cached something somethings?